File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.Deque;
import java.util.ArrayDeque;
import java.util.ArrayList;

class Solution {
    public ArrayList<Integer> solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<Integer>();
        Deque<Integer> q = new ArrayDeque<Integer>();
        
        int  day;
        for (int i = 0; i < speeds.length; i++) {
            day = (int) Math.ceil((double) (100 - progresses[i]) / speeds[i]);
            if (!q.isEmpty() && q.peekFirst() < day) {
                answer.add(q.size());
                q.clear();
            }
            q.addLast(day);
        }
        if (!q.isEmpty()) answer.add(q.size());
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    time = []
    cnt = 0
    for i in range(0, len(speeds)):
        num = math.ceil((100 - progresses[i]) / speeds[i])
        if not time or time[-1][0] < num:
            cnt = 1
            time.append([num, cnt])
        else:
            time[-1][1] += 1
    for t in time:
        answer.append(t[1])
    return answer